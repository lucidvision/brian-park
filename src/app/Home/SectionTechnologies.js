import React from 'react'
import styled from 'styled-components'
import Technologies from './Technologies'
import { HeadingSecondary, Paragraph } from 'components'
import { colors, media, utilities } from 'styles'

const StyledSection = styled.section`
  background-image: linear-gradient(
    to right bottom,
    ${colors.gradientLight},
    ${colors.gradientDark});
  background-size: cover;
  transform: skewY(-7deg);

  & > * {
    transform: skewY(7deg);
  }
`

const StyledHeadingSecondary = styled(HeadingSecondary)`
  text-align: center;
  margin-bottom: ${utilities.marginLarge}

  ${media.phone`
    margin-bottom: ${utilities.marginMedium}
  `}
`

const StyledDiv = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;

  ${media.phone`
    flex-direction: column;
  `}
`

const StyledParagraph = styled(Paragraph)`
  max-width: 50rem;

  ${media.phone`
    margin-bottom: ${utilities.marginMedium}
    text-align: center;
  `}
`

export default function SectionTechnologies (props) {
  return (
    <StyledSection>
      <StyledHeadingSecondary color={colors.white} text='Technologies' />
      <StyledDiv>
        <StyledParagraph color={colors.white} text={'My primary focus is JavaScript. JavaScript\'s ubiquity makes it arguably the most productive programming language. It can be used for web, mobile and APIs. It has cutting-edge tools, libraries and frameworks. Its first-class functions allow for a declarative programming style. JavaScript empowers me to bring products to market fast.'} />
        <Technologies />
      </StyledDiv>
    </StyledSection>
  )
}
