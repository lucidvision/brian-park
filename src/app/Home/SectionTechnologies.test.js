import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import SectionTechnologies from './SectionTechnologies'

Enzyme.configure({adapter: new Adapter()})

describe('SectionTechnologies', () => {
  const sectionTechnologies = shallow(<SectionTechnologies />)

  it('renders correctly', () => {
    expect(sectionTechnologies).toMatchSnapshot()
  })

  it('renders a secondary heading', () => {
    expect(sectionTechnologies.find('Styled(HeadingSecondary)').props().text).toBe('Technologies')
  })

  it('renders a paragraph', () => {
    expect(sectionTechnologies.find('Styled(Paragraph)').props().text).toBe('My primary focus is JavaScript. JavaScript\'s ubiquity makes it arguably the most productive programming language. It can be used for web, mobile and APIs. It has cutting-edge tools, libraries and frameworks. Its first-class functions facilitate a declarative programming style. JavaScript empowers me to bring products to market fast.')
  })

  it('renders technology icons', () => {
    expect(sectionTechnologies.find('BarTechnologies').exists()).toBeTruthy()
  })
})
