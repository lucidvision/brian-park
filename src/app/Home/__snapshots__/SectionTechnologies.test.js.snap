// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionTechnologies renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SectionTechnologies />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(HeadingSecondary)
          color="#fff"
          text="Technologies"
/>,
        <styled.div
          className="wow fadeIn"
>
          <Styled(Paragraph)
                    color="#fff"
                    text="My primary focus is JavaScript. JavaScript's ubiquity makes it arguably the most productive programming language. It can be used for web, mobile and APIs. It has cutting-edge tools, libraries and frameworks. Its first-class functions facilitate a declarative programming style. JavaScript empowers me to bring products to market fast."
          />
          <BarTechnologies />
</styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "#fff",
          "text": "Technologies",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Paragraph)
              color="#fff"
              text="My primary focus is JavaScript. JavaScript's ubiquity makes it arguably the most productive programming language. It can be used for web, mobile and APIs. It has cutting-edge tools, libraries and frameworks. Its first-class functions facilitate a declarative programming style. JavaScript empowers me to bring products to market fast."
/>,
            <BarTechnologies />,
          ],
          "className": "wow fadeIn",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "color": "#fff",
              "text": "My primary focus is JavaScript. JavaScript's ubiquity makes it arguably the most productive programming language. It can be used for web, mobile and APIs. It has cutting-edge tools, libraries and frameworks. Its first-class functions facilitate a declarative programming style. JavaScript empowers me to bring products to market fast.",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(HeadingSecondary)
            color="#fff"
            text="Technologies"
/>,
          <styled.div
            className="wow fadeIn"
>
            <Styled(Paragraph)
                        color="#fff"
                        text="My primary focus is JavaScript. JavaScript's ubiquity makes it arguably the most productive programming language. It can be used for web, mobile and APIs. It has cutting-edge tools, libraries and frameworks. Its first-class functions facilitate a declarative programming style. JavaScript empowers me to bring products to market fast."
            />
            <BarTechnologies />
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": "#fff",
            "text": "Technologies",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Paragraph)
                color="#fff"
                text="My primary focus is JavaScript. JavaScript's ubiquity makes it arguably the most productive programming language. It can be used for web, mobile and APIs. It has cutting-edge tools, libraries and frameworks. Its first-class functions facilitate a declarative programming style. JavaScript empowers me to bring products to market fast."
/>,
              <BarTechnologies />,
            ],
            "className": "wow fadeIn",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "color": "#fff",
                "text": "My primary focus is JavaScript. JavaScript's ubiquity makes it arguably the most productive programming language. It can be used for web, mobile and APIs. It has cutting-edge tools, libraries and frameworks. Its first-class functions facilitate a declarative programming style. JavaScript empowers me to bring products to market fast.",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
